### PR TITLE
Make properties of capabilities more flexible.

### PR DESCRIPTION
**Pull request by mrstegeman**
Also, make InstantaneousPowerProperty for SmartPlug optional.

References mozilla-iot/schemas#2

### CODE REVIEW

File: static/js/components/capability/smart-plug.js [hunk 1]
```diff
@@ -62,12 +62,19 @@ class SmartPlugCapability extends BaseComponent {
...
     this._on = false;
     this._power = 0;
+
     this._onClick = this.__onClick.bind(this);
   }
 
   connectedCallback() {
+    this._havePower =
+      typeof this.dataset.havePower !== 'undefined' ?
+        this.dataset.havePower :
```
Comment (hobinjk) [inline static/js/components/capability/smart-plug.js]: If someone specifies `data-have-power="false"` this evaluates to `"false"` which is truthy (oh Javascript, what a beautiful language)

**Comment by mrstegeman** [comment]
@hobinjk Fixed the one issue. That would be rather convoluted, by I can do it if you think it'll make a big difference. In my mind, all properties from legacy devices will have no `@type`, while things moving forward should (ideally) have a `@type` for each property.

**Comment by hobinjk** [comment]
My only concern would be if someone makes a DIY thing which is a control panel with both a "level" for some potentiometer input and a "brightness" for an LED. It's relatively easy to just tell this hypothetical person that they need to come up with better names though

**Comment by mrstegeman** [comment]
@hobinjk Addressed.

**Comment by hobinjk** [comment]
Great, thanks! LGTM

**Pull request merged**